client/jetpack-connect/authorize.js

```diff
@@ -269,9 +267,21 @@ export class JetpackAuthorize extends Component {
 		return 'sso' === from && isSsoApproved( clientId );
 	}
 
-	isWoo( props = this.props ) {
+	isWooRedirect( props = this.props ) {
 		const { from } = props.authQuery;
-		return includes( [ 'woocommerce-services-auto-authorize', 'woocommerce-setup-wizard' ], from );
+		return includes(
+			[
+				'woocommerce-services-auto-authorize',
+				'woocommerce-setup-wizard',
+				'woocommerce-onboarding',
+			],
+			from
+		);
+	}
+
+	isWooOnboarding( props = this.props ) {
+		const { from } = props.authQuery;
+		return includes( [ 'woocommerce-onboarding' ], from );
```

Comment:
Minor but since we only have one value to compare against, could we simplify to:

`return 'woocommerce-onboarding' === from`?